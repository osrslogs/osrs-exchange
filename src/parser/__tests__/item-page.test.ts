import itemPageConfig from '../../util/test-util/api-item-page-mock.json';
import { ItemDetails } from '../../types';
import parseItemPage from '../item-page';

describe('parseItemPage', () => {
  it('it returns correct details when given valid page', () => {
    const itemDetails: ItemDetails[] = parseItemPage(itemPageConfig.valid);
    expect(itemDetails).toMatchSnapshot();
  });

  it('it returns empty details array when given an empty page', () => {
    const itemDetails: ItemDetails[] = parseItemPage(itemPageConfig.validEmptyItems);
    expect(itemDetails.length).toBe(0);
  });
});
