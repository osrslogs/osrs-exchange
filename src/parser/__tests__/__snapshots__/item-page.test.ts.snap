// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseItemPage it returns correct details when given valid page 1`] = `
Array [
  Object {
    "description": "world",
    "icon": "some-url",
    "id": 1234,
    "members": false,
    "name": "hello",
  },
  Object {
    "description": "world",
    "icon": "some-url",
    "id": 1234,
    "members": true,
    "name": "hello",
  },
]
`;
